File: domain/src/main/java/com/icehousecorp/maunorafiq/domain/forecast/interactor/GetForecast.java
package com.icehousecorp.maunorafiq.domain.forecast.interactor;

import com.icehousecorp.maunorafiq.domain.UseCase;
import com.icehousecorp.maunorafiq.domain.forecast.repository.ForecastRepository;

import javax.inject.Inject;

import rx.Observable;

/**
 * Created by maunorafiq on 11/29/16.
 */

public class GetForecast extends UseCase {

    private String city = "Amsterdam";
    private final ForecastRepository forecastRepository;

    @Inject
    public GetForecast(ForecastRepository forecastRepository) {
        this.forecastRepository = forecastRepository;
    }

    @Override
    protected Observable buildUseCaseObservable() {
        return forecastRepository.forecastWeather(this.city);
    }

    public void setCity(String city) {
        this.city = city;
    }
}

File: domain/src/main/java/com/icehousecorp/maunorafiq/domain/forecast/interactor/GetForecast.kt
package com.icehousecorp.maunorafiq.domain.forecast.interactor

import com.icehousecorp.maunorafiq.domain.UseCase
import com.icehousecorp.maunorafiq.domain.forecast.repository.ForecastRepository

import javax.inject.Inject

import rx.Observable

/**
 * Created by maunorafiq on 11/29/16.
 */

class GetForecast
@Inject
constructor(private val forecastRepository: ForecastRepository) : UseCase() {

    private var city = "Amsterdam"

    override fun buildUseCaseObservable(): Observable<*> {
        return this.forecastRepository.forecastWeather(this.city)
    }

    fun setCity(city: String) {
        this.city = city
    }
}
